src/tunacode/tools/decorators.py

```
@@ -68,47 +82,244 @@ async def wrapper(*args: P.args, **kwargs: P.kwargs) -> R:
+    key_t, value_t = args
+    if key_t is not str:
+        return schema
+
+    additional = _python_type_to_json_schema(value_t)
+    if additional:
+        schema["additionalProperties"] = additional
+    return schema
+
+
+def _schema_for_union_args(args: tuple[object, ...]) -> dict[str, object]:
+    non_none = [a for a in args if a is not type(None)]  # noqa: E721
```

Ruff (0.14.14)

[warning] Unused `noqa` directive (unused: `E721`)

Remove unused `noqa` directive

(RUF100)